lib/grandpa/vote_message.go

```diff
@@ -214,9 +215,11 @@ func (s *Service) validateVoteMessage(from peer.ID, m *VoteMessage) (*Vote, erro
 }
 
 // checkForEquivocation checks if the vote is an equivocatory vote.
-// it returns true if so, false otherwise.
-// additionally, if the vote is equivocatory, it updates the service's votes and equivocations.
-func (s *Service) checkForEquivocation(voter *Voter, vote *SignedVote, stage Subround) bool {
+// If it is an equivocatory vote, the wrapped error `ErrEquivocation` is returned,
+// which can be checked with `error.Is(err, ErrEquivocation)`.
+// Additionally, if the vote is equivocatory, it updates the service's votes and equivocations
+// and reports the equivocation to the runtime.
+func (s *Service) checkForEquivocation(voter *Voter, vote *SignedVote, stage Subround) error {
```

Comment:
rename it to checkAndReportEquivocation

Reply:
Is this preferred? Since it is not always reported (most cases won't be equivocations) and we have a separate function for reporting. Can change still if you but I wasn't sure if this was actually more clear

Reply:
`checkForEquivocation` tells you that it is only checking, thus it should return boolean. The function name should tell that it does more than just check.

Reply:
this makes sense to me! Will update